pandas/core/reshape/concat.py

```diff
@@ -584,7 +594,16 @@ def get_result(self):
                 cons = sample._constructor_expanddim
 
                 index, columns = self.new_axes
-                df = cons(data, index=index, copy=self.copy)
+                mgr = dict_to_mgr(
+                    data,
+                    index,
+                    None,
+                    copy=self.copy,
+                    typ=get_option("mode.data_manager"),
+                )
+                if _using_copy_on_write() and not self.copy:
+                    mgr = mgr.copy(deep=False)
+                df = cons(mgr, copy=self.copy)
```

Comment:
This might be doing a second copy now? (in case of `copy=True`) 

I am also wondering if we could do `mgr = cons(data ...)._mgr` instead. That gives some overhead, but I worry a bit that not using `cons` for parsing the dict of Series might impact behaviour of subclasses. 
(have to think through it for the geopandas case)

Reply:
Good point, set copy to false here